enhancements/insights/container-scanner.md

```diff
@@ -0,0 +1,532 @@
+
+##### Defensive Input and Memory Guarding
+
+While keeping the bulk of analysis logic out of the coordinator process reduces risk, there is still the potential for a deprivileged scanner to influence the coordinator as a corrupted relay through its required communication channel. As an example, a corrupted container could lead a scanner process to write a carefully crafted payload which triggers a buffer overflow and potentially arbitrary code execution in the coordinator. Therefore the coordinator process should take particular care in the handling of all input, assuming potentially hostile data. Additionally, the coordinator process should take advantage of technologies and techniques that mitigate memory corruption defects. As an example, Rust’s memory access model provides strong guarantees of memory safety, and limits state mutability to the smaller set of intended write code paths. Usage of a defensive language like Rust over a traditional memory unsafe language like C would mitigate risk further by providing a second line of defense, as well as preventing a number of common issues frequently leveraged in exploits. 
+
+#### Privacy Risk & Mitigation
+
+As runtime information is extracted from the user containers, it is technically possible to get access to sensitive information that should not be exposed to Red Hat.
+The fingerprints that are executing in the container namespaces must be able to sanitize the extracted information and discard them if they deviate from the expected output.
+
+
+To make sure customer confidentiality is preserved, any new collected data is hashed before they are sent to Red Hat backends. Mechanisms are already in place so that access to the unobfuscated data in Red Hat is granted on a needs-to-know basis.
+
+#### Performance Risk & Mitigation
+
+##### Limited Restrictions for Fingerprints
+
+Since a fingerprint executable runs under a downgraded privilege model, with a limited process life-span, significantly more freedom is afforded to it, but some care is still necessary. Notably, a fingerprint executable can have a larger dependency set beyond the core language SDK to meet its needs. Additionally, fingerprint executables can leverage languages with a greater emphasis on productivity (e.g. Golang), provided the implementations still utilize input and output hardening techniques, and do not load potentially rogue code in the container. Each fingerprint implementation may even have a different language runtime than its peers. The following rules should be followed:
+
+1. __Scanning code should be statically compiled and not rely on dynamic library loading.__ This is essentially already required to achieve stable execution on arbitrary container layers, including those with differing libc implementations, or even scratch containers with no additional libraries
```

Comment:
`should` reads to me like this is optional. Should this not say `must be statically compiled`?

Reply:
fixed, thanks